### PR TITLE
sliding (click dragging) my way (plasma) into her dms (portable generator)

### PR DESCRIPTION
**Pull request by MajManatee**
# Document the changes in your pull request
Allows click dragging of fuel into generators so a borg can power generators.
as you can see this is very unique code that I didn't just steal from molti

# Why is this good for the game?
Generators are used in two situations. One, Everything is fucked and we need power now (so it would improve QOL for everyone affected) and two, we are being silly and doing shenanigans (Same reason, allowing people to be silly is a good thing).

There's no reason to lock QOL stuff arbitrarily.

# Changelog

:cl:   
tweak: generator fuel can be dragged into the generator now.
/:cl:

### CODE REVIEW

File: code/modules/power/port_gen.dm [hunk 1]
```diff
@@ -170,6 +170,12 @@
 /obj/machinery/power/port_gen/pacman/proc/overheat()
 	explosion(src.loc, 2, 5, 2, -1)
 
+/obj/machinery/power/port_gen/pacman/MouseDrop_T(atom/dropping, mob/user)
+	if(is_type_in_typecache(dropping, sheet_path))
```
Comment (iamthesilly) [inline code/modules/power/port_gen.dm]: You should be using istype() here since sheet_path is a typepath and not a list or cache.

**Review by SapphicOverload — CHANGES_REQUESTED** [review]

**Review by SapphicOverload — APPROVED** [review]
Neat

File: code/modules/power/port_gen.dm [hunk 2]
```diff
@@ -170,6 +170,12 @@
 /obj/machinery/power/port_gen/pacman/proc/overheat()
 	explosion(src.loc, 2, 5, 2, -1)
 
+/obj/machinery/power/port_gen/pacman/MouseDrop_T(atom/dropping, mob/user)
+	if(istype(dropping, sheet_path))
```
Comment (iamthesilly) [inline code/modules/power/port_gen.dm]: You forgor to add an Adjacent() check

**Review by SapphicOverload — CHANGES_REQUESTED** [review]

File: code/modules/power/port_gen.dm (hunk 2 above)
Comment (MajManatee) [reply code/modules/power/port_gen.dm]: jesus make up your mind woman

**Review by SapphicOverload — APPROVED** [review]

**Pull request merged**